### PR TITLE
Bug 1980901: Parse GW nil as net zero

### PR DESCRIPTION
**Pull request by jordigilh**
When the GW is returned as `nil` from the agent, it is populated as garbage (`<nil>`) in the json, instead of containing a properly formatted IP. This causes the validation in the service to dump error log messages as it is unable to parse the GW value. 

This fix tackles the parsing part from the agent side, and sends an empty string value when it detects the value of the Gateway is nil.

There is a counterpart PR for the service to handle this use case:

https://github.com/openshift/assisted-service/pull/2223

I also enhanced the unit tests to check that the contents of the processed routes match as expected.

@ori-amizur can you take a look at this PR when you have some time?

### CODE REVIEW

File: src/inventory/interfaces.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func ipWithCidrInCidr(ipWithCidrStr, cidrStr string) bool {
 func analyzeAddress(addr net.Addr) (isIpv4 bool, addrStr string, err error) {
 	ipNet, ok := addr.(*net.IPNet)
 	if !ok {
-		return false, "", fmt.Errorf("Could not cast to *net.IPNet")
+		return false, "", fmt.Errorf("could not cast to *net.IPNet")
```
Comment (ori-amizur) [inline src/inventory/interfaces.go:59]: Is this change necessary?

Comment (jordigilh) [reply src/inventory/interfaces.go:59]: Yes, it is. Errors should not be capitalized, as stated in the golang [coding review document](https://github.com/golang/go/wiki/CodeReviewComments#error-strings), which is being referred in the kuberentes [coding convention document](https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md).
You can also find a warning message in your IDE by the go-staticcheck tool:
`error strings should not be capitalized (ST1005)`

**Comment by jordigilh** [comment]
@ori-amizur can you approve if you have no additional comments? Thanks.
